src/bots/HypernymBot.py

```diff
@@ -0,0 +1,61 @@
+from .BotBase import BotBase
+import random
+import itertools
+import fasttext.util
+from nltk.corpus import wordnet as wn
+from scipy.spatial import distance
+import numpy as np
+
+
+class HypernymBot(BotBase):
+    def __init__(self, is_captain=False, is_team_member=False, language="es"):
+        super().__init__(is_captain, is_team_member, language)
+        self.__model = self.__load_model()
+
+    def give_clue(self):
+        return f"{'red' if self.is_red else 'blue'}clue {random.randint(2,5)}"
+
+    def give_answer(self, clue):
+        clue = str.lower(clue)
+        triple_list = list(map(lambda word: (word, self.__get_first_common_hypernym(clue, word)), self._board.get_available_words()))
+        triple_list.sort(key = lambda w: -(np.sum(w[1]) + 0.001*(1/(1+distance.cosine(self.__model.get_word_vector(w[0]), self.__model.get_word_vector(clue))))))
+        return triple_list[0][0]
+
+    def __get_hypernyms_of_synsets(self, synsets):
+        hypernyms = list(map(lambda x: x.hypernyms(), synsets))
+        return list(set(itertools.chain(*hypernyms)))
+
+    def __get_path_similarity(self, synset1, synset2):
+        similarity = wn.path_similarity(synset1, synset2)
+        similarity = similarity if similarity else 0
+        return similarity
+
+
+    def __get_first_common_hypernym(self, word1, word2, lang = 'spa'):
+        try:
```

Comment:
I don't want to review this code. Please, refactor this code using better names, methods, and other techniques. Then we can talk.